sys/reboot/src/log_reboot.c

```diff
@@ -175,41 +184,93 @@ log_reboot_write(const struct log_reboot_info *info)
     }
 #endif
 
-    rc = imgr_read_info(boot_current_slot, &ver, hash, NULL);
+    rc = imgr_read_info(boot_current_slot, &ver, hash, &flags);
     if (rc != 0) {
         return rc;
     }
 
-    off = 0;
-    off += snprintf(buf + off, sizeof buf - off,
-                    "rsn:%s, cnt:%u, img:%u.%u.%u.%u, hash:",
-                    REBOOT_REASON_STR(info->reason), reboot_cnt, ver.iv_major,
-                    ver.iv_minor, ver.iv_revision,
-                    (unsigned int)ver.iv_build_num);
+    memset(cbor_enc_buf, 0, sizeof cbor_enc_buf);
+
+    cbor_buf_writer_init(&writer, cbor_enc_buf, sizeof cbor_enc_buf);
+    cbor_encoder_init(&enc, &writer.enc, 0);
+    rc = cbor_encoder_create_map(&enc, &map, CborIndefiniteLength);
+    if (rc != 0) {
+        return rc;
+    }
+
+    cbor_encode_text_stringz(&map, "rsn");
+    cbor_encode_text_stringz(&map, REBOOT_REASON_STR(info->reason));
 
+    cbor_encode_text_stringz(&map, "cnt");
+    cbor_encode_int(&map, reboot_cnt);
+
+    cbor_encode_text_stringz(&map, "img");
+    snprintf(buf, sizeof buf, "%u.%u.%u.%u",
+                  ver.iv_major, ver.iv_minor, ver.iv_revision,
+                  (unsigned int)ver.iv_build_num);
+    cbor_encode_text_stringz(&map, buf);
+
+    cbor_encode_text_stringz(&map, "hash");
+    memset(buf, 0, sizeof buf);
```

Comment:
this is not really needed snprintf will work fine

Reply:
Since we are talking lot of hypothetical things, what if hash is empty?

Reply:
what goes into string is ascii representation of hex encoded bytes, sizeof hash is not 0, so buf is always filled with something, and it should be null terminated after every round

Reply:
Ahh I see, I overlooked what we are using here is sizeof for looping, then it makes sense to assume buf will be null terminated.